tests/integration/test_http_handlers_config/test_predefined_handler/config.xml

```diff
@@ -33,5 +33,35 @@
                 <query>INSERT INTO test_table(id, data) SELECT {id:UInt32}, {_request_body:String}</query>
             </handler>
         </rule>
+
+        <rule>
+            <methods>GET</methods>
+            <url>/test_predefined_handler_auth_with_password</url>
+            <handler>
+                <type>predefined_query_handler</type>
+                <user>with_password</user>
+                <password>password</password>
```

Comment:
I don't understand why we need to specify the user's password in the configuration file.

First of all, it is insecure. Secondly, it is unneeded because the server can assume this user internally. We can even allow creating users without any way to authenticate externally.

Reply:
The idea of this PR is to set that different default users should be used depending on which rule is chosen in `<clickhouse.http_handlers>` while connecting to ClickHouse.

I agree it isn't secure to specify passwords in the server configuration file. Probably we can leave only user names here in `<clickhouse.http_handlers>` and add some clause to `CREATE USER` to specify somehow that the user with no password can be used only if a specific http handler is used.

Reply:
>The idea of this PR is to set that different default users should be used depending on which rule is chosen in <clickhouse.http_handlers> while connecting to ClickHouse.

Actually the initial goal was to just hide from users authorization

I.e. imagine I'm exporting some metric via `http_handlers`, now I have to give the user some credentials that it could use to access the data, and also I have to `GRANT` required permission too the shared user.

>Probably we can leave only user names here in <clickhouse.http_handlers>

Agree

> and add some clause to CREATE USER to specify somehow that the user with no password can be used only if a specific http handler is used.

I was thinking about making this users works like inter-server secret, that does not require password for user at all.